deltachat-ios/AppDelegate.swift

```diff
@@ -372,38 +372,46 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         bgIoTimestamp = nowTimestamp
 
         // make sure to balance each call to `beginBackgroundTask` with `endBackgroundTask`
+        let fetchSemaphore = DispatchSemaphore(value: 0)
         let backgroundTask = UIApplication.shared.beginBackgroundTask {
-            // TODO: currently, this should be okay, we are not using too much background time,
-            // so that handler should never be called.
-            // the plan is to listen to an event as DC_EVENT_ENTER_IDLE and stop fetch from there.
-            // if we have such an event, we could imprive this handler and fire the event.
-            logger.info("fetch background task will end soon")
+            // we do not call `endBackgroundTask` here, even if that results in a warning:
+            // if we would do, we would risk a 0xdead10cc exception as things are still running;
+            // therefore we just signal the task below to exit gracefully and call `endBackgroundTask`.
+            // in practise, this handler should not be called anyway
+            // as we take care to stay below the 30 seconds.
```

Comment:
calling the handler itself would be easy, just needs a var in local namespace, declared with `var backgroundTask: UIBackgroundTaskIdentifier = .invalid` - however, it is unclear if that is finally better.